docs/examples/reusable-components.md

```diff
@@ -0,0 +1,262 @@
+# Reusable Components
+
+In this example, we illustrate how the creation of reusable UI components works using `ff4s`. Two of many,
+button and select components are popular in frontend development and are going to be used in this example along with the [Frankfurter API](https://frankfurter.app).
+
+## Components
+
+In this example, custom select and button components are implemented as member methods of a `Components` class
+with type parameters `F` for the effect type, `S` for the state type and `A` for the action type.
+
+### Select
+
+A select component with options of type `O` is modelled by a function with the following inputs:
+
+1. `fromString`: converts a string option to type `O`.
+2. `onChange0`: performs an action based on selected option and state.
+3. `options`: list of options.
+4. `selected0`: currently selected option function of the state.
+
+### Button
+
+Likewise for a button component inputs:
+
+1. `onClick0`: react to click events by performing an action.
+2. `isDisabled`: disable the button based on the state.
+
+```scala mdoc:js:shared
+import org.scalajs.dom
+import cats.Show
+import cats.kernel.Eq
+import cats.syntax.all._
+
+// Class defining all components
+class Components[F[_], S, A] {
+
+  // Custom select component
+  def customSelect[O: Show: Eq](
+      fromString: String => Option[O],
+      onChange0: (S, O) => Option[A],
+      options: List[O],
+      selected0: S => O
+  )(implicit dsl: ff4s.Dsl[F, S, A]): dsl.V = {
+    import dsl._
+    import dsl.html._
+    useState { state =>
+      select(
+        onChange := ((ev: dom.Event) =>
+          ev.target match {
+            case el: dom.HTMLSelectElement =>
+              fromString(el.value).flatMap(onChange0(state, _))
+            case _ => None
+          }
+        ),
+        options.map { name =>
+          option(
+            selected := (name === selected0(state)), // use of Eq[O] implicit
+            key := name.show, // use of Show[O] implicit
+            value := name.show,
+            name.show
+          )
+        }
+      )
+    }
```

Comment:
We can use pristine argument names and rename the imports instead
```suggestion
  def customSelect[O: Show: Eq](
      fromString: String => Option[O],
      onChange: (S, O) => Option[A],
      options: List[O],
      selected: S => O
  )(implicit dsl: ff4s.Dsl[F, S, A]): dsl.V = {
    import dsl._
    import dsl.html.{onClick => onClick0, selected => selected0, _}
    useState { state =>
      select(
        onChange0 := ((ev: dom.Event) =>
          ev.target match {
            case el: dom.HTMLSelectElement =>
              fromString(el.value).flatMap(onChange(state, _))
            case _ => None
          }
        ),
        options.map { name =>
          option(
            selected0 := (name === selected(state)),
            key := name.show,
            value := name.show,
            name.show
          )
        }
      )
    }
```

Reply:
nice one!